### DISCUSSION TITLE
Experimenting with graph and charts output

### DISCUSSION THREAD

**Discussion by Freddeb** [sinaptik-ai/pandas-ai#1480] [General]
### System Info

**System info: pandasai version 2.2.15, Windows 10, Python 3.12.2, Azure OpenAI model  gpt-4o-mini**

Hi,
I'm testing the library (By the way, great work!). I would like please to have your feedback on some tests that I've done.
My test consists to read an Excel file with sales data (this file comes from internet and it's a free sample).

Here are the questions that I asked to the library and the output results:
**Test 1:**
in_user_prompt="For the country Belgium, plot a graph with the total revenue on Y axis and with the ship date on X axis. Draw a line with the red color."

**Test 2**
in_user_prompt="For the country Albania, plot a graph with the total revenue on Y axis and with the ship date on X axis. Draw a line with the blue color."

**Test 3**
in_user_prompt="For the country Belgium and Albania, plot a graph with the total revenue on Y axis with the ship date on X axis in the ship date period between 2011 and 2016. For Albania draw the line with the color blue, for Belgium use the color red."

I've joined my script and the Excel file to this thread.
[sample_sales_1000.xlsx](https://github.com/user-attachments/files/17462496/sample_sales_1000.xlsx)
[excel_test.txt](https://github.com/user-attachments/files/17462587/excel_test.txt)

Thank you for your feedback.
Fred

### 🐛 Describe the bug

Test 1 results:
![image](https://github.com/user-attachments/assets/d9677b67-b5ef-4387-8028-4c7fe7128339)

Test 2 results:
![image](https://github.com/user-attachments/assets/c2e7ee24-7d3c-4160-84cd-1629376ae682)
Remark: The graph displays a blue dot for each measure (the dots do not appear in the first graph - Test 1). 

Test 3 results:
![image](https://github.com/user-attachments/assets/14550689-98e7-4e8a-bd0c-a39ad076498b)
Remark: The graph displays a discontinued blue line for the country Albania without a dot for each measure.  The country Belgium has a small segment (probably because the period covers between 2011 and 2016).

**Comment by gdcsinaptik** [comment]
Hey @Freddeb super interesting. Moving this ticket to a github discussion, in the general challenge, so that we can hopefully get some conversation going on what you want to achieve and how other users have done it :)
